code/app/src/main/java/com/kernelcrew/moodapp/data/Comment.java

```diff
@@ -0,0 +1,88 @@
+package com.kernelcrew.moodapp.data;
+
+import android.graphics.Bitmap;
+
+import androidx.annotation.Nullable;
+
+import com.google.firebase.firestore.Exclude;
+import com.kernelcrew.moodapp.utils.PhotoUtils;
+
+import java.io.Serializable;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * A logged mood event owned by a user referenced by that user's firebase auth UID.
+ */
+public class Comment implements Serializable {
+    private String id;
+    private String uid;
+    private String username;
+    private String moodEventId;
+    private Date created;
+    private String commentText;
+
+
+    /**
+     * Empty constructor for Firestore deserialization. Do not use.
+     */
+    public Comment() { }
+
+    /**
+     * Constructor for a new Comment with additional details.
+     * Will assign this comment a new random id.
+     */
+    public Comment(String uid, String username, String moodEventId, String commentText) {
+        this.id = UUID.randomUUID().toString();
+        this.uid = uid;
+        this.username = username;
+        this.moodEventId = moodEventId;
+        this.created = new Date();
+        this.commentText = commentText;
+    }
+
+    public String logging() {
```

Comment:
Can just be `toString()`. That would make more sense. Otherwise, we should add a JavaDoc.